com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbePlacement.cs

```diff
@@ -3,236 +3,443 @@
+                            for (int i = 0; i < readbackBrickCount; i++)
+                            {
+                                var pos = ctx.brickPositions[i];
+                                var brick = new Brick(new Vector3Int(Mathf.RoundToInt(pos.x), Mathf.RoundToInt(pos.y), Mathf.RoundToInt(pos.z)), localSubdivLevel);
+                                bricksSet.Add(brick);
+                            }
+                        }
+                    });
                 }
+
+                cmd.WaitAllAsyncReadbackRequests();
+                Graphics.ExecuteCommandBuffer(cmd);
+
+                finalBricks = bricksSet.ToList();
+
+                // TODO: this is really slow :/
+                Profiler.BeginSample($"Sort {finalBricks.Count} bricks");
+                // sort from larger to smaller bricks
+                finalBricks.Sort((Brick lhs, Brick rhs) =>
+                {
+                    if (lhs.subdivisionLevel != rhs.subdivisionLevel)
+                        return lhs.subdivisionLevel > rhs.subdivisionLevel ? -1 : 1;
+                    if (lhs.position.z != rhs.position.z)
+                        return lhs.position.z < rhs.position.z ? -1 : 1;
+                    if (lhs.position.y != rhs.position.y)
+                        return lhs.position.y < rhs.position.y ? -1 : 1;
+                    if (lhs.position.x != rhs.position.x)
+                        return lhs.position.x < rhs.position.x ? -1 : 1;
+
+                    return 0;
+                });
+                Profiler.EndSample();
             }
+            Profiler.EndSample();
 
-            return num;
+            return finalBricks;
         }
 
-        static protected void CullVolumes(ref List<ProbeReferenceVolume.Volume> cullees, ref List<ProbeReferenceVolume.Volume> cullers, ref List<ProbeReferenceVolume.Volume> result)
+        static void RastersizeMeshes(CommandBuffer cmd, ProbeReferenceVolume.Volume cellVolume, RenderTexture sceneSDF, RenderTexture dummyRenderTarget, int maxBrickCountPerAxis, List<(Renderer component, ProbeReferenceVolume.Volume volume)> renderers)
         {
-            foreach (ProbeReferenceVolume.Volume v in cullers)
+            using (new ProfilingScope(cmd, new ProfilingSampler("Rasterize Meshes 3D")))
             {
-                ProbeReferenceVolume.Volume lv = v;
+                var cellAABB = cellVolume.CalculateAABB();
 
-                foreach (ProbeReferenceVolume.Volume c in cullees)
+                using (new ProfilingScope(cmd, new ProfilingSampler("Clear")))
                 {
-                    if (result.Contains(c))
-                        continue;
+                    cmd.SetComputeTextureParam(subdivideSceneCS, s_ClearKernel, _Output, sceneSDF);
+                    DispatchCompute(cmd, s_ClearKernel, sceneSDF.width, sceneSDF.height, sceneSDF.volumeDepth);
+                }
+
+                cmd.SetRandomWriteTarget(4, sceneSDF);
```

Comment:
0 doesn't work, 1 and 2 crash the editor (Dx11), and 4 isn't used by HDRP or URP IIRC.
This binding thing is really weird, I don't think I can do something better than that, I'll put a name instead of this magic number though